numba/tests/test_analysis.py

```diff
@@ -563,6 +564,238 @@ def impl(array, x=None, a=None):
                           np.zeros((2, 3)), 1.2, None)
 
 
+class TestBranchPrunePredicates(TestBranchPruneBase, SerialMixin):
+    # Really important thing to remember... the branch on predicates end up as
+    # POP_JUMP_IF_<bool> and the targets are backwards compared to normal, i.e.
+    # the true condition is far jump and the false the near i.e. `if x` would
+    # end up in Numba IR as e.g. `branch x 10, 6`.
+
+    _TRUTHY = (1, "String", True, 7.4, 3j)
+    _FALSEY = (0, "", False, 0.0, 0j, None)
+
+    def _literal_const_sample_generator(self, pyfunc, consts):
```

Comment:
I spent sometime thinking of alternative ways to do this. I can't think of anything less scary.

Reply:
This pattern is used so much in testing now I'm tempted to make it a utility!

Reply:
right, the bits that does the `if utils.PYVERSION >= (3, 8):` should not be duplicated too many times.